chaosaws/ec2/actions.py

```diff
@@ -59,13 +59,14 @@ def stop_instances(instance_ids: List[str] = None, az: str = None,
 
     client = aws_client('ec2', configuration, secrets)
 
-    if az:
-        filters = [{'Name': 'availability-zone', 'Values': [az]}]
-        instance_ids = list_instance_ids(filters, client)
+    if not instance_ids:
+        if az:
+            filters = [{'Name': 'availability-zone', 'Values': [az]}]
+            instance_ids = list_instance_ids(filters, client)
 
-        if not instance_ids:
-            raise FailedActivity(
-                "No instances in availability zone: {}".format(az))
+            if not instance_ids:
```

Comment:
Might be your editor change the indentation but I'd rather we stay consistant, the indent should be as I had set it I believe :)